resources/fetching/fetchers/azure/assets_enricher.go

```diff
@@ -36,6 +36,7 @@ func initEnrichers(provider azurelib.ProviderAPI) []AssetsEnricher {
 	enrichers = append(enrichers, vmNetworkSecurityGroupEnricher{})
 	enrichers = append(enrichers, sqlServerEnricher{provider: provider})
 	enrichers = append(enrichers, postgresqlEnricher{provider: provider})
+	enrichers = append(enrichers, keyVaultEnricher{provider: provider})
```

Comment:
Circling back to the discussion we had a few weeks ago, we said that in case we had too many asset enrichers, we would consider creating a package for it, just for directory structure purposes.

What do you think? I'm pro creating a new package.

(If you agree, let's do it in another PR soon)

Reply:
My concern, in general (when thinking about creating a new package), is whether we need an isolated package or not (no matter the number of files that are in it): functionality-wise, or/and if we need to hide some implementation or/and if the new package offers some kind of specialization (like for example `encoding` and `encoding/json` where the second specializes and depends on the first)

We could create a package `azure/enrichers`, but if the only reason for that is the number of files, I would be a bit skeptical (not against though) . 

Let's decide async for a next possible PR, agreed.